python/az/aro/azext_aro/custom.py

```diff
@@ -273,13 +273,20 @@ def generate_random_id():
     return random_id
 
 
-def get_route_tables_from_subnets(cli_ctx, subnets):
+def get_route_tables_from_subnets(cli_ctx, subnets, fail):
     network_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_NETWORK)
 
     route_tables = set()
     for sn in subnets:
         sid = parse_resource_id(sn)
 
+        if not all(key in sid.keys() for key in ('resource_group', 'name', 'resource_name')):
+            if fail:
+                logger.error("Failed to validate subnet '%s'", sn)
+                logger.warning("Please retry, if issue presists: raise azure support ticket")
+                raise ValidationError(f"(ValidationError) Failed to validate subnet '{sn}'")
```

Comment:
nit: it is generally an anti pattern to have multiple log statements for the same 'error' as well as to log and raise an exception, can we combine this? Especially since the error log is the same text as the exception.

typo: presists -> persists
```suggestion
            if fail:
                raise ValidationError(f"(ValidationError) Failed to validate subnet '{sn}'. Please retry, if issue persists: raise azure support ticket")
```